dmaws/rds.py

```diff
@@ -295,20 +287,27 @@ def clean_database(self):
                     WHERE (
                         SELECT COUNT(*) FROM supplier_frameworks WHERE supplier_id=suppliers.supplier_id
                     ) = 0
+                    AND supplier_id NOT IN (
+                        SELECT DISTINCT supplier_id FROM services
+                    )
                 ), d1 AS (
                     DELETE FROM contact_information WHERE supplier_id IN (SELECT supplier_id FROM dangling_suppliers)
                 ), d2 AS (
                     DELETE FROM users WHERE supplier_id IN (SELECT supplier_id FROM dangling_suppliers)
                 )
-             DELETE FROM suppliers WHERE supplier_id IN (SELECT supplier_id FROM dangling_suppliers)
+            DELETE FROM suppliers WHERE supplier_id IN (SELECT supplier_id FROM dangling_suppliers)
         """)
 
+        # Remove audit events because we don't use them
```

Comment:
I'm not sure... if we have no "changed" audit events we should still be able to test the service editing flow - it will should just treat the current version of services as fresh and new.  Feels like a pain to keep just certain events.  And the draft edit date ones I think only make sense during a live procurement, so those drafts shouldn't be copied over anyway.  

I just remember seeing some pages break when those audit events have been deleted.